src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// is negative.
+    /// The index is zero-based, that is, using index 0 returns the first element.
     /// </summary>
-    /// <exception cref="ArgumentException">Thrown when the sequence has insufficient length or
-    /// <paramref name="index" /> is negative.</exception>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The nth element of the task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    /// <exception cref="T:ArgumentException">Thrown when the sequence has insufficient length or <paramref name="index" /> is negative.</exception>
     val item: index: int -> source: taskSeq<'T> -> Task<'T>
 
     /// <summary>
     /// Returns the only element of the task sequence, or <see cref="None" /> if the sequence is empty of
     /// contains more than one element.
     /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The only element of the singleton task sequence, or <see cref="None" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     val tryExactlyOne: source: taskSeq<'T> -> Task<'T option>
 
     /// <summary>
     /// Returns the only element of the task sequence.
     /// </summary>
-    /// <exception cref="ArgumentException">Thrown when the input sequence does not contain precisely one element.</exception>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The only element of the singleton task sequence, or <see cref="None" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    /// <exception cref="T:ArgumentException">Thrown when the input task sequence does not contain precisely one element.</exception>
     val exactlyOne: source: taskSeq<'T> -> Task<'T>
 
     /// <summary>
     /// Applies the given function <paramref name="chooser" /> to each element of the task sequence. Returns
-    /// a sequence comprised of the results "x" for each element where
-    /// the function returns <c>Some(x)</c>.
+    /// a sequence comprised of the results where the function returns <see cref="Some(x)" />.
     /// If <paramref name="chooser" /> is asynchronous, consider using <see cref="TaskSeq.chooseAsync" />.
     /// </summary>
+    ///
+    /// <param name="chooser">A function to transform items of type <typeref name="'T" /> into options of type <typeref name="'U" />.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     val choose: chooser: ('T -> 'U option) -> source: taskSeq<'T> -> taskSeq<'U>
```

Comment:
aside: any thoughts on voption overloads? I guess they can come if/when overloading becomes possible? Or is all that stuff best left alongside valuetasks etc as a big future thing (to not do)?

Reply:
@bartelink Yeah, I have been considering this, and similarly for the `Async<_>` overloads (i.e. apart from the name, the `XXXAsync` functions take a `Task`).

I hope to soon make the switch to static members.